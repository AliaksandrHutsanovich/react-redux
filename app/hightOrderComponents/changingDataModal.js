import React, { useState, memo } from 'react';
import 'antd/dist/antd.css';
import { Modal, Input } from 'antd';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { clearReDo } from '../actions/actions';
import { typesCategoryOperation } from './utils/utils';

function changingDataDialog(operationTitle) {
  const ChangingDataDialog = ({
    dispatch,
    title,
    visible,
    handleOk,
    titleCategory,
    handleCancel,
    path,
  }) => {
    const [value, setValue] = useState('');

    const handleChange = (e) => {
      setValue(e.target.value);
    };

    const handleClickOk = () => {
      dispatch(clearReDo());
      typesCategoryOperation[operationTitle](path, dispatch, value, title);
      setValue('');
      handleOk();
    };

    return (
      <Modal
        title={operationTitle}
        visible={visible}
        onOk={() => handleClickOk(handleOk, path, dispatch, title)}
        onCancel={handleCancel}
      >
        {
          operationTitle === 'Delete category'
            ? <p>{titleCategory}</p>
            : (
              <Input
                placeholder="input category title"
                defaultValue={title}
                onChange={handleChange}
                className="modal-input"
              />
            )
        }
      </Modal>
    );
  };

  ChangingDataDialog.defaultProps = {
    title: '',
    titleCategory: '',
  };

  ChangingDataDialog.propTypes = {
    dispatch: PropTypes.func.isRequired,
    title: PropTypes.string,
    visible: PropTypes.bool.isRequired,
    handleOk: PropTypes.func.isRequired,
    path: PropTypes.string.isRequired,
    titleCategory: PropTypes.string,
    handleCancel: PropTypes.func.isRequired,
  };

  return connect()(memo(ChangingDataDialog));
}

export default changingDataDialog;
