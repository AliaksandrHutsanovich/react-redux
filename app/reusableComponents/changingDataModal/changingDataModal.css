.modalInput {
    width: 100%;
}

.button {
    top: 10px;
    left: 410px;
}
