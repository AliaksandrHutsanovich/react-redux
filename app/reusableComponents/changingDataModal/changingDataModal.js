import React, { useState, memo } from 'react';
import 'antd/dist/antd.css';
import { Modal, Input } from 'antd';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { clearReDo } from '../../actions/actions';
import { typesCategoryOperation } from '../../hightOrderComponents/utils/utils';

import styles from './changingDataModal.css';

const ChangingDataDialog = ({
  dispatch,
  title,
  visible,
  handleOk,
  titleCategory,
  handleCancel,
  path,
  operationTitle,
}) => {
  const [value, setValue] = useState('');

  const handleChange = (e) => {
    setValue(e.target.value);
  };

  const handleClickOk = () => {
    dispatch(clearReDo());
    typesCategoryOperation[operationTitle](path, dispatch, value, title);
    setValue('');
    handleOk();
  };

  return (
    <Modal
      title={operationTitle}
      visible={visible}
      onOk={() => handleClickOk(handleOk, path, dispatch, title)}
      onCancel={handleCancel}
    >
      {
        operationTitle === 'Delete category'
          ? <p>{titleCategory}</p>
          : (
            <Input
              placeholder="input category title"
              defaultValue={title}
              onChange={handleChange}
              className={styles.modalInput}
            />
          )
        }
    </Modal>
  );
};

ChangingDataDialog.defaultProps = {
  title: '',
  titleCategory: '',
};

ChangingDataDialog.propTypes = {
  dispatch: PropTypes.func.isRequired,
  title: PropTypes.string,
  visible: PropTypes.bool.isRequired,
  handleOk: PropTypes.func.isRequired,
  path: PropTypes.string.isRequired,
  titleCategory: PropTypes.string,
  handleCancel: PropTypes.func.isRequired,
  operationTitle: PropTypes.string.isRequired,
};

export default connect()(memo(ChangingDataDialog));
