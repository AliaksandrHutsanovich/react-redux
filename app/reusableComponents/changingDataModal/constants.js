import OPERATION_TITLES from '../../constants';

const FOOTER_BY_OPERATION_TITLE = {
  [OPERATION_TITLES.EDIT_CATEGORY]: [],
  [OPERATION_TITLES.ADD_NEW_SUBCATEGORY]: [],
};

export default FOOTER_BY_OPERATION_TITLE;
