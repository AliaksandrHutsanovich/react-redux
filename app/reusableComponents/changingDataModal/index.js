export { default, ChangingDataDialog } from './changingDataModal';
