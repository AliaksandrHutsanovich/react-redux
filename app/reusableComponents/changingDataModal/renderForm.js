import React from 'react';
import { Controller } from 'react-hook-form';
import { Input, Typography } from 'antd';

import styles from './changingDataModal.css';

const { Text } = Typography;

const renderForm = (
  categoryTitle,
  handleClickOk,
  handleSubmit,
  control,
  errors,
  formId,
) => (
  <form
    id={formId}
    autoComplete="off"
    onSubmit={handleSubmit(({ title }) => handleClickOk(title))}
  >
    <Controller
      as={(
        <Input
          placeholder="input category title"
          className={styles.modalInput}
        />
      )}
      rules={{ required: 'This is required' }}
      control={control}
      name="title"
      defaultValue={categoryTitle}
    />
    {
      errors.title
        && (
          <Text
            type="danger"
          >
            {errors.title.message}
          </Text>
        )
    }
  </form>
);

export default renderForm;
