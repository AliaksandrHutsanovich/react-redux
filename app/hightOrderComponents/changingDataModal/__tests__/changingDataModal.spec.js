import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import changingDataModal from '../changingDataModal';

const mockStore = configureStore();
const initialState = {};
const CustomChangingDataModal = changingDataModal('Add new subcategory');

describe('Unit tests of add changing data dialog', () => {
  const store = mockStore(initialState);
  const Component = mount(
    <CustomChangingDataModal
      store={store}
      title="category"
      visible
      handleOk={() => {}}
      handleCancel={() => {}}
      path="/category"
    />,
  );

  it('full render test', () => {
    expect(Component).toMatchSnapshot();
  });

  it('input should be changable', () => {
    Component.find('input').simulate(
      'change',
      {
        target: {
          value: 'javascript',
        },
      },
    );
    expect(Component.find('input').props().value).toEqual('javascript');
  });

  it('button Ok should be clickable', () => {
    Component.find('button').at(2).simulate('click');
  });
});
