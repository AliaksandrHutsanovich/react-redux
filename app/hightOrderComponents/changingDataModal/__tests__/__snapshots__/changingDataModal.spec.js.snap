// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests of add changing data dialog full render test 1`] = `
<Connect(Component)
  handleCancel={[Function]}
  handleOk={[Function]}
  path="/category"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  title="category"
  visible={true}
>
  <ChangingDataDialog
    dispatch={[Function]}
    handleCancel={[Function]}
    handleOk={[Function]}
    path="/category"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    title="category"
    visible={true}
  >
    <Modal
      cancelButtonDisabled={false}
      confirmLoading={false}
      maskTransitionName="fade"
      okButtonDisabled={false}
      okType="primary"
      onCancel={[Function]}
      onOk={[Function]}
      title="Add new subcategory"
      transitionName="zoom"
      visible={true}
      width={520}
    >
      <DialogWrap
        cancelButtonDisabled={false}
        closeIcon={
          <span
            className="ant-modal-close-x"
          >
            <Icon
              className="ant-modal-close-icon"
              type="close"
            />
          </span>
        }
        confirmLoading={false}
        footer={
          <LocaleReceiver
            componentName="Modal"
            defaultLocale={
              Object {
                "cancelText": "Cancel",
                "justOkText": "OK",
                "okText": "OK",
              }
            }
          >
            [Function]
          </LocaleReceiver>
        }
        forceRender={false}
        maskTransitionName="fade"
        okButtonDisabled={false}
        okType="primary"
        onCancel={[Function]}
        onClose={[Function]}
        onOk={[Function]}
        prefixCls="ant-modal"
        title="Add new subcategory"
        transitionName="zoom"
        visible={true}
        width={520}
        wrapClassName=""
      >
        <Portal
          getContainer={[Function]}
        >
          <Portal
            containerInfo={
              <div>
                <div>
                  <div
                    class="ant-modal-mask fade-appear"
                  />
                  <div
                    aria-labelledby="rcDialogTitle0"
                    class="ant-modal-wrap "
                    role="dialog"
                    tabindex="-1"
                  >
                    <div
                      class="ant-modal zoom-appear"
                      role="document"
                      style="width: 520px;"
                    >
                      <div
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      >
                        sentinelStart
                      </div>
                      <div
                        class="ant-modal-content"
                      >
                        <button
                          aria-label="Close"
                          class="ant-modal-close"
                        >
                          <span
                            class="ant-modal-close-x"
                          >
                            <i
                              aria-label="icon: close"
                              class="anticon anticon-close ant-modal-close-icon"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="close"
                                fill="currentColor"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                />
                              </svg>
                            </i>
                          </span>
                        </button>
                        <div
                          class="ant-modal-header"
                        >
                          <div
                            class="ant-modal-title"
                            id="rcDialogTitle0"
                          >
                            Add new subcategory
                          </div>
                        </div>
                        <div
                          class="ant-modal-body"
                        >
                          <input
                            class="ant-input modalInput"
                            placeholder="input category title"
                            type="text"
                            value="category"
                          />
                        </div>
                        <div
                          class="ant-modal-footer"
                        >
                          <div>
                            <button
                              class="ant-btn"
                              type="button"
                            >
                              <span>
                                Cancel
                              </span>
                            </button>
                            <button
                              class="ant-btn ant-btn-primary"
                              type="button"
                            >
                              <span>
                                OK
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                      <div
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      >
                        sentinelEnd
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            }
          >
            <Dialog
              cancelButtonDisabled={false}
              className=""
              closable={true}
              closeIcon={
                <span
                  className="ant-modal-close-x"
                >
                  <Icon
                    className="ant-modal-close-icon"
                    type="close"
                  />
                </span>
              }
              confirmLoading={false}
              destroyOnClose={false}
              footer={
                <LocaleReceiver
                  componentName="Modal"
                  defaultLocale={
                    Object {
                      "cancelText": "Cancel",
                      "justOkText": "OK",
                      "okText": "OK",
                    }
                  }
                >
                  [Function]
                </LocaleReceiver>
              }
              forceRender={false}
              key="dialog"
              keyboard={true}
              mask={true}
              maskClosable={true}
              maskTransitionName="fade"
              okButtonDisabled={false}
              okType="primary"
              onCancel={[Function]}
              onClose={[Function]}
              onOk={[Function]}
              prefixCls="ant-modal"
              title="Add new subcategory"
              transitionName="zoom"
              visible={true}
              width={520}
              wrapClassName=""
            >
              <div>
                <Animate
                  animation={Object {}}
                  component=""
                  componentProps={Object {}}
                  key="mask"
                  onAppear={[Function]}
                  onEnd={[Function]}
                  onEnter={[Function]}
                  onLeave={[Function]}
                  showProp="visible"
                  transitionAppear={true}
                  transitionEnter={true}
                  transitionLeave={true}
                  transitionName="fade"
                >
                  <AnimateChild
                    animation={Object {}}
                    key="mask"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="fade"
                  >
                    <LazyRenderBox
                      className="ant-modal-mask"
                      hiddenClassName="ant-modal-mask-hidden"
                      key="mask"
                      style={Object {}}
                      visible={true}
                    >
                      <div
                        className="ant-modal-mask"
                        style={Object {}}
                      />
                    </LazyRenderBox>
                  </AnimateChild>
                </Animate>
                <div
                  aria-labelledby="rcDialogTitle0"
                  className="ant-modal-wrap "
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                  tabIndex={-1}
                >
                  <Animate
                    animation={Object {}}
                    component=""
                    componentProps={Object {}}
                    key="dialog"
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="visible"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="zoom"
                  >
                    <AnimateChild
                      animation={Object {}}
                      key="dialog-element"
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="zoom"
                    >
                      <LazyRenderBox
                        className="ant-modal "
                        key="dialog-element"
                        role="document"
                        style={
                          Object {
                            "width": 520,
                          }
                        }
                        visible={true}
                      >
                        <div
                          className="ant-modal "
                          role="document"
                          style={
                            Object {
                              "width": 520,
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          >
                            sentinelStart
                          </div>
                          <div
                            className="ant-modal-content"
                          >
                            <button
                              aria-label="Close"
                              className="ant-modal-close"
                              onClick={[Function]}
                            >
                              <span
                                className="ant-modal-close-x"
                              >
                                <Icon
                                  className="ant-modal-close-icon"
                                  type="close"
                                >
                                  <LocaleReceiver
                                    componentName="Icon"
                                  >
                                    <i
                                      aria-label="icon: close"
                                      className="anticon anticon-close ant-modal-close-icon"
                                    >
                                      <IconReact
                                        className=""
                                        type="close-o"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="close"
                                          fill="currentColor"
                                          height="1em"
                                          key="svg-close"
                                          viewBox="64 64 896 896"
                                          width="1em"
                                        >
                                          <path
                                            d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                            key="svg-close-svg-0"
                                          />
                                        </svg>
                                      </IconReact>
                                    </i>
                                  </LocaleReceiver>
                                </Icon>
                              </span>
                            </button>
                            <div
                              className="ant-modal-header"
                            >
                              <div
                                className="ant-modal-title"
                                id="rcDialogTitle0"
                              >
                                Add new subcategory
                              </div>
                            </div>
                            <div
                              className="ant-modal-body"
                            >
                              <Input
                                className="modalInput"
                                defaultValue="category"
                                disabled={false}
                                onChange={[Function]}
                                placeholder="input category title"
                                type="text"
                              >
                                <input
                                  className="ant-input modalInput"
                                  disabled={false}
                                  onChange={[Function]}
                                  onKeyDown={[Function]}
                                  placeholder="input category title"
                                  type="text"
                                  value="category"
                                />
                              </Input>
                            </div>
                            <div
                              className="ant-modal-footer"
                            >
                              <LocaleReceiver
                                componentName="Modal"
                                defaultLocale={
                                  Object {
                                    "cancelText": "Cancel",
                                    "justOkText": "OK",
                                    "okText": "OK",
                                  }
                                }
                              >
                                <div>
                                  <Button
                                    block={false}
                                    ghost={false}
                                    loading={false}
                                    onClick={[Function]}
                                  >
                                    <Wave>
                                      <button
                                        className="ant-btn"
                                        onClick={[Function]}
                                        type="button"
                                      >
                                        <span
                                          key=".0"
                                        >
                                          Cancel
                                        </span>
                                      </button>
                                    </Wave>
                                  </Button>
                                  <Button
                                    block={false}
                                    ghost={false}
                                    loading={false}
                                    onClick={[Function]}
                                    type="primary"
                                  >
                                    <Wave>
                                      <button
                                        className="ant-btn ant-btn-primary"
                                        onClick={[Function]}
                                        type="button"
                                      >
                                        <span
                                          key=".0"
                                        >
                                          OK
                                        </span>
                                      </button>
                                    </Wave>
                                  </Button>
                                </div>
                              </LocaleReceiver>
                            </div>
                          </div>
                          <div
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          >
                            sentinelEnd
                          </div>
                        </div>
                      </LazyRenderBox>
                    </AnimateChild>
                  </Animate>
                </div>
              </div>
            </Dialog>
          </Portal>
        </Portal>
      </DialogWrap>
    </Modal>
  </ChangingDataDialog>
</Connect(Component)>
`;
